tapir/shifts/views/calendars.py

```diff
@@ -136,12 +141,28 @@ def get_context_data(self, **kwargs):
 class ShiftTemplateGroupCalendar(LoginRequiredMixin, TemplateView):
     template_name = "shifts/shift_template_group_calendar.html"
 
-    def get_context_data(self, **kwargs):
-        context = super().get_context_data()
-        date_to_group = {}
-        today = timezone.now().date()
-        for week in range(52):
-            monday = today - timedelta(days=today.weekday()) + timedelta(weeks=week)
-            date_to_group[monday] = get_week_group(monday).name
-        context["date_to_group"] = date_to_group
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+
+        # TODO Frederik: a static calendar.html would accelerate
+        thisyear = date.today().year
+        start_date = date(thisyear, 1, 1)
+        end_date = date(thisyear, 12, 31)
+        delta = end_date - start_date  # returns timedelta
+        shift_dict = {}
+        for i in range(delta.days + 1):
+            # iterate through days of year
+            day = start_date + datetime.timedelta(days=i)
+            monday = day - datetime.timedelta(days=day.weekday() % 7)
+            if monday not in shift_dict.keys():
+                # populate with shift names at first day of week
+                shift_dict[monday] = get_week_group(monday).name
+        cal = ColorHTMLCalendar(firstweekday=MONDAY, shift_dict=shift_dict)
+        html_result = cal.formatyear(theyear=thisyear, width=4)
+        # TODO relative paths: HOWTO?
+        HTML(string=html_result).write_pdf(
+            "tapir/shifts/static/shifts/ABCD_weeks_calendar.pdf",
```

Comment:
We should not write to the filesystem. We cannot assume that it is writable and that static content is actually served from there. Instead, we should have a separate view that serves the calendar as a PDF.

Reply:
Good idea, I changed that